src/python_testing/TC_G_2_3.py

```diff
@@ -0,0 +1,382 @@
+            endpoint=0,
+            attribute=Clusters.GroupKeyManagement.Attributes.GroupTable
+        )
+        matching_entry = next((entry for entry in groupTableList if entry.groupId == kGroupId6), None)
+        asserts.assert_is_not_none(matching_entry, f"GroupId 0x{kGroupId6:04X} not found in GroupTable attribute")
+
+        self.step("11b")
+        if await self.feature_guard(endpoint=self.matter_test_config.endpoint, cluster=Clusters.Groups, feature_int=Clusters.Groups.Bitmaps.Feature.kGroupNames):
+            group_found = False
+            for group in groupTableList:
+                if group.groupName == kGroupNameGp6:
+                    group_found = True
+                    break
+            asserts.assert_true(group_found, f"Group with name '{kGroupNameGp6}' not found in the GroupTable")
+        else:
+            self.mark_current_step_skipped()
+
+        self.step("12")
+        kGroupId7 = 7
+        kGroupNameGp7 = "Gp7"
+        resp = await th1.SendCommand(self.dut_node_id, self.matter_test_config.endpoint, Clusters.Groups.Commands.AddGroupIfIdentifying(kGroupId7, kGroupNameGp7))
+
+        self.step("13")
+        groupTableList: List[Clusters.GroupKeyManagement.Attributes.GroupTable] = await self.read_single_attribute(
+            dev_ctrl=th1, node_id=self.dut_node_id, endpoint=0, attribute=Clusters.GroupKeyManagement.Attributes.GroupTable)
+        expected_endpoint = self.matter_test_config.endpoint
+        received_groups = [(entry.groupId, entry.endpoints) for entry in groupTableList]
+        asserts.assert_true(
+            (kGroupId7, [expected_endpoint]) in [(groupId, enps) for groupId, enps in received_groups if expected_endpoint in enps],
+            f"Expected GroupId {kGroupId7} with endpoint {expected_endpoint} not found in DUT GroupTable: {received_groups}"
+        )
+
+        self.step("14")
+        if await self.feature_guard(endpoint=self.matter_test_config.endpoint, cluster=Clusters.Groups, feature_int=Clusters.Groups.Bitmaps.Feature.kGroupNames):
+            group_found = False
+            for group in groupTableList:
+                if group.groupName == kGroupNameGp7:
+                    group_found = True
+                    break
+            asserts.assert_true(group_found, f"Group with name '{kGroupNameGp7}' not found in the GroupTable")
+        else:
+            self.mark_current_step_skipped()
+
+        self.step("15")
+        startGroupId = 8
+        groupList = [startGroupId + i for i in range(maxgroups - 2)]
+        for kGroupId in groupList:
+            await th1.SendCommand(self.dut_node_id, self.matter_test_config.endpoint, Clusters.Groups.Commands.AddGroupIfIdentifying(kGroupId, f"Gp{kGroupId}"))
+
+        self.step("16a")
+        GroupKeyMapStruct = Clusters.GroupKeyManagement.Structs.GroupKeyMapStruct
+        groupKeyMapStruct = [GroupKeyMapStruct(groupId=i, groupKeySetID=kGroupKeySetID) for i in range(6+maxgroups, 6+maxgroups+1)]
```

Comment:
how is this working out? range is not inclusive of the end (https://docs.python.org/2/library/functions.html#range) so this is setting a single value - 6+maxgroups.

so how are you getting resource exhaused? Also, 6+maxgroups below is not a different group ID

Can you please check the logic in this test and ensure things are working as expected? Is this related to the incorrect range in step 1b?